clients/venice-client/src/main/java/com/linkedin/venice/fastclient/InternalAvroStoreClient.java

```diff
@@ -53,11 +97,14 @@ protected CompletableFuture<Map<K, V>> batchGet(BatchGetRequestContext<K, V> req
     streamingResultFuture.whenComplete((response, throwable) -> {
       if (throwable != null) {
         resultFuture.completeExceptionally(throwable);
-      } else if (!requestContext.isPartialSuccessAllowed && requestContext.getPartialResponseException().isPresent()) {
-        resultFuture.completeExceptionally(
-            new VeniceClientException("Response was not complete", requestContext.getPartialResponseException().get()));
       } else {
-        resultFuture.complete(response);
+        Throwable partialResponseException = checkBatchGetPartialFailure(requestContext);
```

Comment:
Originally, I was thinking that `RetriableAvroGenericStoreClient` should decide whether the request with retry enabled succeed or not and other layers should be just pass-through.
Can you explain why we need to invoke this function here?